### PR TITLE
Handle the edxapp worker case with two ASGs but no ELBs

### PR DESCRIPTION
**Pull request by jibsheet**
Previously it would exit without a message but no inventory.

### CODE REVIEW

File: playbooks/active_instances_in_asg.py [hunk 1]
```diff
@@ -61,6 +62,12 @@ def run(self,asg_name):
                 # When we have more than a single active ASG, we need to bail out as we don't know what ASG to pick an instance from
                 print("Multiple active ASGs - unable to choose an instance", file=sys.stderr)
                 return
+            # If we found no active groups, because there are no ELBs (edxapp workers normally)
+            # print a sensible reason why we failed
+            elbs = list(chain.from_iterable([group['LoadBalancerNames'] for group in matching_groups]))
+            if len(active_groups) == 0 and len(elbs) == 0:
```
Comment (jibsheet) [inline playbooks/active_instances_in_asg.py]: More pythonic ways to express lines 67/68 would be interesting

**Comment by jibsheet** [comment]
@edx/devops please review
FYI @rlucioni

File: playbooks/active_instances_in_asg.py (hunk 1 above)
Comment (rlucioni) [reply playbooks/active_instances_in_asg.py]: For line 68:
```py
if not (active_groups or elbs):
    ...
```

Comment (jibsheet) [reply playbooks/active_instances_in_asg.py]: Thanks, I don't know why I was so fixated on using len()

**Review by fredsmith — APPROVED** [review]

**Review by feanil — APPROVED** [review]
This implies that we can't use this inventory script to run tasks on the workers.  Is the plan to use ec2.py in the cases where we need to run things against the workers?

**Comment by jibsheet** [comment]
@feanil We can use it on the workers, we only get in here if there is more than 1 ASG.  There's a ticket up for grooming to write something we can use to properly target workers.  This was a patch to try and give them a better error message while we fix the underlying issue.

**Pull request merged**